crates/uv-build-backend/src/source_dist.rs

```diff
@@ -33,8 +34,14 @@ pub fn build_source_dist(
         extension: SourceDistExtension::TarGz,
     };
     let source_dist_path = source_dist_directory.join(filename.to_string());
-    let writer = TarGzWriter::new(&source_dist_path)?;
+
+    let temp_file = NamedTempFile::new_in(source_dist_directory)?;
+    let writer = TarGzWriter::new(temp_file.as_file().try_clone()?, &source_dist_path);
```

Comment:
Rather than `temp_file.as_file().try_clone()?` just make `TarGzWriter::new` accept `&temp_file` (not really fussed between `&'a NamedTempFile` and `writer: W where W: Write`, it's only used by this function).

I don't know what cases `try_clone` would fail in but it's better to just avoid the opportunity entirely.